### PR TITLE
some filesystem command signatures

### PR DESCRIPTION
**Pull request by raccmonteiro**
# Description

Signature for the following commands:

- `cd` 
- `cp` 
- `glob` 
- `ls` 
- `mkdir` 
- `mv` 
- `open`
- `rm`
- `save`
- `touch`

Related to https://github.com/nushell/nushell/issues/7320

### CODE REVIEW

**Comment by sholderbach** [comment]
Weird that the test runner didn't start on this one.

Input refers to what is on the left side of the `| command`. So I don't think that `Type::String` matches for `cd` or `ls`.
I am unsure if we currently allow specifying different output types for one single input type. In this case I think you have to loosen the requirements to `Type::Any` or what ever matches the description.

**Comment by raccmonteiro** [comment]
> Weird that the test runner didn't start on this one.
> 
> Input refers to what is on the left side of the `| command`. So I don't think that `Type::String` matches for `cd` or `ls`. I am unsure if we currently allow specifying different output types for one single input type. In this case I think you have to loosen the requirements to `Type::Any` or what ever matches the description.

Oh I was considering command parameters is also an input, I see now that this is wrong. I will update the code, thanks for the explanation.

**Comment by raccmonteiro** [comment]
I wanted to reply and closed this by mistake :( sorry

**Comment by sholderbach** [comment]
Thanks for the quick turnaround! 

> About multiple outputs, in `mkdir` if we use the `--verbose` flag we get a list of created directories, it's better to have a single output with `Type::Any` or `Type::List(Box::new(Type::String))` ? Can we consider it's an empty list when the flag `-v` is not activated?

Making it an empty list sounds like a good compromise, as long as it isn't polluting the downstream output. For the coercion and checking rules @dandavison is probably the foremost expert.

File: crates/nu-command/src/filesystem/open.rs [hunk 1]
```diff
@@ -35,6 +36,7 @@ impl Command for Open {
 
     fn signature(&self) -> nu_protocol::Signature {
         Signature::build("open")
+            .input_output_types(vec![(Type::String, Type::Nothing)])
```
Comment (sholderbach) [inline crates/nu-command/src/filesystem/open.rs:39]: This should always output `Type::Any` and has currently two variants in that it accepts either path like input or nothing.

**Comment by sholderbach** [comment]
Thanks again for your work. Let's move that forward!

**Comment by raccmonteiro** [comment]
I forgot I had this unfinished, thanks @sholderbach for fixing some mistakes

**Comment by sholderbach** [comment]
No problem, you got us 99% there so I didn't want to bother you any further.

**Pull request merged**